internal/services/pubkey_service_test.go

```diff
@@ -0,0 +1,71 @@
+package services
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/RHEnVision/provisioning-backend/internal/dao"
+	"github.com/RHEnVision/provisioning-backend/internal/models"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	firstPub = models.Pubkey{
+		ID:        1,
+		AccountID: 2,
+		Name:      "firstkey",
+		Body:      "sha-rsa body",
+	}
+)
+
+type PubkeyStoreMock struct{}
+
+func (mock PubkeyStoreMock) GetPubkeyDao(context.Context) (dao.PubkeyDao, error) {
+	return mock, nil
+}
+
+func (PubkeyStoreMock) Create(ctx context.Context, pk *models.Pubkey) error { return nil }
+func (PubkeyStoreMock) Update(ctx context.Context, pk *models.Pubkey) error { return nil }
+func (PubkeyStoreMock) GetById(ctx context.Context, id uint64) (*models.Pubkey, error) {
+	return &firstPub, nil
+}
+func (PubkeyStoreMock) List(ctx context.Context, limit, offset uint64) ([]*models.Pubkey, error) {
+	secondPub := models.Pubkey{
+		ID:        2,
+		AccountID: 4,
+		Name:      "secondkey",
+		Body:      "sha-rsa body",
+	}
+	mockedPubkeys := []*models.Pubkey{&firstPub, &secondPub}
+
+	return mockedPubkeys, nil
+}
+func (PubkeyStoreMock) Delete(ctx context.Context, id uint64) error { return nil }
+
+func TestListPubkeysHandler(t *testing.T) {
+	GetPubkeyDao = (&PubkeyStoreMock{}).GetPubkeyDao
+
+	req, err := http.NewRequestWithContext(context.TODO(), "GET", "/api/provisioning/pubkeys", nil)
+
+	assert.Nil(t, err, fmt.Sprintf("Error creating a new request: %v", err))
+
+	rr := httptest.NewRecorder()
+	handler := http.HandlerFunc(ListPubkeys)
+	handler.ServeHTTP(rr, req)
+
+	if status := rr.Code; status != http.StatusOK {
+		t.Errorf("Handler returned wrong status code. Expected: %d. Got: %d.", http.StatusOK, status)
+	}
+
+	var pubkeys []models.Pubkey
+
+	if err := json.NewDecoder(rr.Body).Decode(&pubkeys); err != nil {
+		t.Errorf("Error decoding response body: %v", err)
+	}
+
+	assert.Equal(t, 2, len(pubkeys), "expected two pubkeys in response json")
+}
```

Comment:
This almost feels like an integration test, I think you want to stub DAO instead and then test this service unchanged.

Reply:
I believe this is minimum required to test that the method does what it is supposed to :) Thus Unit test.
Integration test should not IMHO skip even the server.